modules/nodes-cluster-resource-configure-jdk.adoc

```diff
@@ -29,29 +29,19 @@ this documentation, and may involve setting multiple additional JVM options.
 == Understanding how to override the JVM maximum heap size
 
 For many Java workloads, the JVM heap is the largest single consumer of memory.
-Currently, the OpenJDK defaults to allowing up to 1/4 (1/`-XX:MaxRAMFraction`)
-of the compute node's memory to be used for the heap, regardless of whether the
-OpenJDK is running in a container or not. It is therefore *essential* to
-override this behavior, especially if a container memory limit is also set.
 
-There are at least two ways the above can be achieved:
+Currently, on bare metal or VM setups, the OpenJDK's default heap value is 25%, 1/4th of the compute node’s memory. However, within containers using cgroups detection (v1 and v2 for JDK 11/17), Xmx was set as 50% of the max memory resource limits by default. In ubi9 - RHEL 9 images, the default allocation is 80%(set via -XX:MaxRAMPercentage), leaving 20% for native use. It is therefore essential to
```

Comment:
THere's a lot of historical stuff here that I think can go. We now have full cgroups support in all JDK shipped by by RHEL (JDK8, 11 and newer).  I think it's important to clearly differentiate the behaviour of OpenJDK from the Red Hat OpenJDK containers, since  not all Java-using customers will be using the latter. How about this?

----

OpenJDK’s default heap value is 25% or <sup>1</sup>/<sub>4</sub><sup>th</sup> of
either the compute node’s memory or, if specified, the container memory limit.

The Red Hat UBI OpenJDK containers default to 80% of compute node memory or
container memory limit.

If this is not an appropriate value for your deployment, it can be overridden
with the command-line flag `-XX:MaxRAMPercentage`. For the Red Hat UBI OpenJDK
containers, this can be specified by setting the environment variable
`JAVA_MAX_MEM_RATIO` to the desired value e.g. `JAVA_MAX_MEM_RATIO=50`.

----

Additional context: 80% has been chosen as a sensible default for in-containers. If a user has specified a container memory limit, then it's a reasonable assumption that they want the JVM to use  virtually all of that memory, but we need to leave a little headroom for any non-java processes that might run in the same container (e.g. liveness probes and suchlike ,or things spawned by tools like `oc cp`). If the user has NOT set a memory limit, then we need to be careful that the JVM doesn't take all of the node's RAM, as this is very unlikely to be what they want. 80% is thought to be the right balance.

Reply:
I'd actually delete everything else that follows in this file (except the header for the next section on the last line): notes about UseContainerSupport are historic, defining MaxRAM etc is not recommended if MaxRAMPercentage can achieve what is needed.

Reply:
What Jon suggested sounds good to me!

Reply:
Okay..... Let me delete the note.